modules/govuk_elasticsearch/manifests/init.pp

```diff
@@ -106,13 +121,18 @@
   }
 
   elasticsearch::instance { $::fqdn:
+    notify        => Exec['secure_es_yaml'],
     config        => $instance_config_real,
     datadir       => '/mnt/elasticsearch',
     init_defaults => {
       'ES_HEAP_SIZE' => $heap_size,
     },
   }
 
+  exec { 'secure_es_yaml':
+    command => "chmod 0640 /etc/elasticsearch/${::fqdn}/elasticsearch.yml",
+  }
```

Comment:
I'm also not sure it works - the state of the system when Puppet has finished running is this:

```
-rw-r--r-- 1 elasticsearch elasticsearch 845 May 16 12:13 /etc/elasticsearch/logs-elasticsearch-1.management.dev.gov.uk/elasticsearch.yml
```